packages/client/test/db.e2e.spec.ts

```diff
@@ -0,0 +1,243 @@
+} from '@blockful/contracts/out/UniversalResolver.sol/UniversalResolver.json'
+
+import { ethers as eth } from 'hardhat'
+import { Contract } from 'ethers'
+import { normalize, labelhash, namehash } from 'viem/ens'
+import { localhost } from 'viem/chains'
+import { createTestClient, http, publicActions } from 'viem'
+import { HardhatEthersSigner } from '@nomicfoundation/hardhat-ethers/signers'
+import { expect } from 'chai'
+
+import { NewApp } from '@blockful/gateway/src/app'
+import {
+  withGetAddr,
+  withGetContentHash,
+  withGetText,
+} from '@blockful/gateway/src/handlers'
+import { InMemoryRepository } from '@blockful/gateway/src/repositories'
+import { withSigner } from '@blockful/gateway/src/middlewares'
+
+const gatewayUrl = 'http://127.0.0.1:3000/{sender}/{data}.json'
+
+let dbResolver: Contract,
+  registry: Contract,
+  universalResolver: Contract,
+  signers: HardhatEthersSigner[]
+
+const client = createTestClient({
+  chain: localhost,
+  mode: 'hardhat',
+  transport: http(),
+}).extend(publicActions)
+
+async function deployOffchainResolver() {
+  const ResolverContract = await new eth.ContractFactory(
+    abiOffchainResolver,
+    bytecodeOffchainResolver,
+    signers[0],
+  ).deploy(gatewayUrl, signers)
+
+  dbResolver = await eth.getContractAt(
+    abiOffchainResolver,
+    await ResolverContract.getAddress(),
+  )
+}
+
+async function deployRegistry() {
+  const RegistryContract = await new eth.ContractFactory(
+    abiRegistry,
+    bytecodeRegistry,
+    signers[0],
+  ).deploy()
+
+  registry = await eth.getContractAt(
+    abiRegistry,
+    await RegistryContract.getAddress(),
+  )
+
+  await registry.setSubnodeRecord(
+    eth.ZeroHash,
+    labelhash('eth'),
+    signers[0],
+    await dbResolver.getAddress(),
+    10000000,
+  )
+  await registry.setSubnodeRecord(
+    namehash('eth'),
+    labelhash('offchain'),
+    signers[0],
+    await dbResolver.getAddress(),
+    10000000,
+  )
+}
+
+async function deployUniversalResolver() {
+  const UniversalResolverContract = await new eth.ContractFactory(
+    abiUniversalResolver,
+    bytecodeUniversalResolver,
+    signers[0],
+  ).deploy(await registry.getAddress(), [gatewayUrl])
+
+  universalResolver = await eth.getContractAt(
+    abiUniversalResolver,
+    await UniversalResolverContract.getAddress(),
+  )
+}
+
+function setupGateway(
+  privateKey: `0x${string}`,
+  { repo }: { repo: InMemoryRepository },
+) {
+  const app = NewApp(
+    [withGetText(repo), withGetAddr(repo), withGetContentHash(repo)],
+    [
+      withSigner(privateKey, [
+        'function text(bytes32 node, string key)',
+        'function addr(bytes32 node)',
+        'function contenthash(bytes32 node)',
+      ]),
+    ],
+  )
+  app.listen('3000')
+}
+
+describe('DatabaseResolver', () => {
+  let repo: InMemoryRepository
+  const rawNode = 'database.eth'
+  const node = namehash(rawNode)
+  const domains = new Map()
+  const domain = {
+    node,
+    ttl: 99712622115,
```

Comment:
Does this value have something special? Why this number?

Reply:
just a number that the local blockchain will never reach